tests/unit/v2/cache/testMemoryBackend.js

```diff
@@ -0,0 +1,106 @@
+const assert = require('assert');
+const { MemoryCache } = require('../../../../libV2/cache').backends;
+const { shardFromTimestamp } = require('../../../../libV2/utils');
+const schema = require('../../../../libV2/cache/schema');
+
+const { generateFakeEvents } = require('../../../utils/v2Data');
+
+const startTime = new Date().getTime();
+const endTime = startTime + 30000; // Add 30 seconds;
+const testValues = generateFakeEvents(startTime, endTime, 50);
+
+describe('Test cache memory backend', () => {
```

Comment:
Is there a way to use fixtures to test multiple backends in the same way? As this test code looks similar to the Redis backend.

Reply:
I considered it, but decided that given the direct introspection of the db I in the memory backend that it would lead to some sort of wrapper to abstract the asserts that would have to be maintained as well. Since the tested API is small and the tests relatively short I think the added cost of abstraction would outweigh the maintaining separate tests.